### ISSUE TITLE
Nested populate for a document that is not defined returns empty array

### ISSUE THREAD

**Issue by kfrankovic** [Automattic/mongoose#6798]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Let's say we have following models: Driver, Car and Producer (car producer). Driver contains array of references to Car model, car has reference (single reference, not an array!) to Producer. Both references are not required. 

In first step we populate array of Cars in Drivers, array of references will correctly be substituted with Car objects. Then we again populate same list of Drivers, but this time we populate Producers in previously populated array of Cars. If car reference was defined, Car will now correctly contain Producer object, but if Car reference is missing (to be precise - producer property in Car object is undefined) it will be substituted with **empty array**. We expected that they would still be undefined after populated is executed.

It also seems that this happens only if at least one Car has reference to Producers. 

In our case this can be problematic because in lot of our conditions instead of `cars[0].producer` being evaluated as false (since they were undefined), they are now evaluated as true (since they contain empty array). 

We detected this bug yesterday after update to version 5.2.5. This bug also appears in 5.2.6. We found out that this bug was introduced with version 5.0.12. and assume that his happens because of following fix:

https://github.com/Automattic/mongoose/issues/6245
https://github.com/Automattic/mongoose/commit/9570a1b5f30ad09f7b62821ed185dba412aaa491

(we confirmed that this bug doesn't appear in 5.0.11, but appears in 5.0.12 and above).

Easiest way to reproduce and understand issue is script bellow. It is consistently reproducing issue with appropriate mongoose versions. 

**If the current behavior is a bug, please provide the steps to reproduce.**

```
const mongoose = require('mongoose');
mongoose.connect('mongodb://localhost:27017/issue_test');

const DriverSchema = new mongoose.Schema({
  name: 'String',
  cars: [{ type: 'ObjectId', required: false, ref: 'Car' }]
})

const CarSchema = new mongoose.Schema({ 
  name: 'String',
  producer: { type: 'ObjectId', required: false, ref: 'Producer'}
});

const ProducerSchema = new mongoose.Schema({
  name: 'String',
})

const Driver = mongoose.model('Driver', DriverSchema);
const Car = mongoose.model('Car', CarSchema);
const Producer = mongoose.model('Producer', ProducerSchema);

async function execute() {
  const oneProducer = new Producer({ name: 'Producer A '});
  await oneProducer.save();

  const oneCar = new Car({ name: 'My personal car', producer: oneProducer });
  await oneCar.save();

  const oneDriver = new Driver({ name: 'Driver 1', cars: [oneCar]})
  await oneDriver.save();

  const secondCar = new Car({ name: 'Work car' });
  await secondCar.save();

  const secondDriver = new Driver({ name: 'Driver 2', cars: [secondCar] });
  await secondDriver.save();

  const drivers = await Driver.find().populate('cars');

  const _drivers = await Driver.populate(drivers, {
    path: 'cars.producer',
    model: 'Producer',
  })

  console.log(_drivers[1].cars[0].producer);
  // Mongoose version 5.0.11 will print 'undefined'
  // Mongoose versions >= 5.0.12 will print '[]' (empty array)
}

execute();
```

Package.json just for easier running:

```
{
  "name": "reproduce",
  "version": "0.0.1",
  "description": "Reproducer mongo bug",
  "main": "index.js",
  "scripts": {
    "start": "node index.js"
  },
  "dependencies": {
    "mongoose": "5.2.6"
  }
}
```

**What is the expected behavior?**

In above scenario - populating missing property should still result in that field being undefined after populate.

**Please mention your node.js, mongoose and MongoDB version.**

Node: 8.10.0
Mongoose: 5.2.6
MongodDB: v3.4, but we've also seen this behavior on v3.2

**Comment by vkarpov15** [comment]
Thanks for reporting, will fix ASAP.

**Comment by djambrecina** [comment]
@vkarpov15 The issue is fixed for the given example above. But it still persists for the more complex example where one schema is actually the third schema in the populate chain. Although, I'm not sure if this is the correct way to use mongoose populates.
Please take a look at the following example:
```const mongoose = require('mongoose');
mongoose.connect('mongodb://localhost:27017/issue_test');

const FileSchema = new mongoose.Schema({
  filename: {
    type: 'String'
  }
});
const EditionOptionsSchema = new mongoose.Schema({
  price: {
    type: 'Number'
  },
  image: {
    type: 'ObjectId',
    ref: 'File'
  }
});
const EditionSchema = new mongoose.Schema({
  released: {
    type: 'Boolean',
  },
  editionOptions: {
    type: EditionOptionsSchema,
  }
});
const CarVersionSchema = new mongoose.Schema({
  type: {
    type: 'String'
  },
  editions: [{
    type: 'ObjectId',
    ref: 'Edition'
  }]
});
const CarSchema = new mongoose.Schema({
  make: 'String',
  versions: [CarVersionSchema]
});

const File = mongoose.model('File', FileSchema);
const Edition = mongoose.model('Edition', EditionSchema);
const Car = mongoose.model('Car', CarSchema);

async function cleanup() {
  await File.remove();
  await Edition.remove();
  await Car.remove();
}

async function setup() {
  const file1 = new File({ filename: 'file1.png' });
  await file1.save();

  const edition1 = new Edition({
    released: true,
    editionOptions: {
      price: 100000,
      image: file1
    }
  });
  await edition1.save();

  const car1 = new Car({
    make: 'Mercedes',
    versions: [{
      type: 'C Class',
      editions: [edition1]
    }]
  });
  await car1.save();

  const edition2 = new Edition({
    released: true,
    editionOptions: {
      price: 80000
    }
  });
  await edition2.save();

  const car2 = new Car({
    make: 'Ford',
    versions: [{
      type: 'Focus',
      editions: [edition2]
    }]
  });
  await car2.save();
}

async function fetch() {
  let cars = await Car.find();

  cars = await Car.populate(cars, {
    path: 'versions.editions',
		model: 'Edition',
		options: { lean: true }
  });

  cars = await Car.populate(cars, {
		path: 'versions.editions.editionOptions.image',
		model: 'File',
		options: { lean: true }
	});

  console.log(cars[1].versions[0].editions[0].editionOptions.image);
  // Mongoose version 5.0.11 will print 'undefined'
  // Mongoose versions >= 5.0.12 will print '[]' (empty array)
}

async function execute() {
  await cleanup();
  await setup();
  await fetch();
}

execute();
```

**Comment by vkarpov15** [comment]
Thanks for reporting, will investigate asap

**Comment by vkarpov15** [comment]
Fixed in master with a01b827fddcf4f35ea62ed30e4c99a5106f199d5
